### PR TITLE
Linux: Add cachestat, fchmodat2 syscalls

### PR DESCRIPTION
**Pull request by The-King-of-Toasters**
I haven't wired up fchmodat2 yet, as it's pretty fresh and I wasn't sure to put it in with the fchmodat impl or in its own file.

### CODE REVIEW

File: test/libc/calls/cachestat_test.c [hunk 1]
```diff
@@ -0,0 +1,119 @@
+/*-*- mode:c;indent-tabs-mode:nil;c-basic-offset:2;tab-width:8;coding:utf-8 -*-│
+│vi: set net ft=c ts=2 sts=2 sw=2 fenc=utf-8                                :vi│
+╞══════════════════════════════════════════════════════════════════════════════╡
+│ Copyright 2022 Justine Alexandra Roberts Tunney                              │
```
Comment (jart) [inline test/libc/calls/cachestat_test.c:4]: [optional] This is test code. Feel free to make your mark in this file rather than attributing it to me.

**Review by jart — CHANGES_REQUESTED** [review]
Thanks for sending this! System calls are cosmo's bread and butter. It's very exciting to see contributions helping improve them.

File: libc/sysv/syscalls.sh [hunk 2]
```diff
@@ -384,6 +384,8 @@ scall	sys_memfd_secret	0xfffffffffffff1bf	0xfff	globl # no wrapper
 scall	sys_process_mrelease	0xfffffffffffff1c0	0xfff	globl # no wrapper
 scall	sys_futex_waitv		0xfffffffffffff1c1	0xfff	globl # no wrapper
 scall	sys_set_mempolicy_home_node 0xfffffffffffff1c2	0xfff	globl # no wrapper
+scall	sys_cachestat		0xfffffffffffff1c3	0x1c3	globl # Linux 6.5+
+scall	sys_fchmodat2		0xfffffffffffff1c4	0x1c4	globl # no wrapper Linux 6.6+
```
[the remark this answers is not in the document]
Comment (The-King-of-Toasters) [reply libc/sysv/syscalls.sh:388]: I wasn't sure if I should use `cosmo_once` for this ala `copy_file_range`.

[the remark this answers is not in the document]
Comment (jart) [reply libc/sysv/syscalls.sh:388]: I'm not terribly concerned about the performance of `fchmodat`.

**Comment by The-King-of-Toasters** [comment]
I tried adding `cachestat` to `kPledgeStdio`, but I kept getting ENOSYS errors. Might be worth following up yourself.

**Review by jart — APPROVED** [review]
Looks good. Thank you!

**Pull request merged**